stock_account_move_multi_reset_to_draft/models/account_move.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+from odoo.tools.float_utils import float_is_zero
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def button_draft(self):
+        """If it is a purchase invoice, we will create a new SVL for each line with
+        the sum of the value in opposite sign.
+        """
+        res = super().button_draft()
+        for item in self.filtered(
+            lambda x: x.is_inbound
+            and any(line.stock_valuation_layer_ids for line in x.sudo().line_ids)
+        ):
+            for line in item.sudo().line_ids.filtered("stock_valuation_layer_ids"):
```

Comment:
Try this to call sudo only once for all lines.
```suggestion
        for item in self.sudo().filtered(
            lambda x: x.is_inbound
            and any(line.stock_valuation_layer_ids for line in x.line_ids)
        ):
            for line in item.line_ids.filtered("stock_valuation_layer_ids"):
```